### DISCUSSION TITLE
trim start end make duration vagued

### DISCUSSION THREAD

**Discussion by sugizo** [livingbio/typed-ffmpeg#318] [Q&A] [answered]
**code**
```
import ffmpeg
import IPython
in_file = ffmpeg.input('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4')
ffmpeg.output(in_file.trim(start = 5, end = 8),
              in_file.atrim(start = 5, end = 8),
              filename = "trim_atrim_time.mp4", ).overwrite_output().run()
IPython.display.Video('trim_atrim_time.mp4', embed = True)
```

**result not expected**
video duration is 8 secs

**expected result**
video duration is 3 secs 
which explain on
8 (end) - 5 (start) = 3

**code just atrim (for audio)**
```
import ffmpeg
import IPython
in_file = ffmpeg.input('[BigBuckBunny](http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny).mp4')
ffmpeg.output(in_file.atrim(start = 5, end = 8),
              filename = "trim_atrim_time.mp3", ).overwrite_output().run()
IPython.display.Audio('trim_atrim_time.mp3')
```

**result is expected**
audio duration is 3 secs

problem seems on trim(start, end) function

**question**
how to trim video 
with exact start 
and 
end time 
using typed-ffmpeg ?

thanks and best regards

**Comment by lucemia** [comment, marked as answer]
For more information, please see:
https://www.bannerbear.com/blog/how-to-trim-a-video-using-ffmpeg/

It's advisable to avoid using a complex filter for simple video trimming tasks:

```python
import ffmpeg

ffmpeg.input(
    "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
    ss=5,
    t=3,
).output(filename="trim_without_complex_filter.mp4").overwrite_output().run()
```

https://github.com/livingbio/typed-ffmpeg/assets/432851/c3f82cf8-0574-4d97-8998-6c728e54a7c8


When using a complex filter for trimming, sometimes the `setpts` filter becomes necessary:

> Slightly different from the last two commands, setpts=PTS-STARTPTS is added to the command to adjust the timestamps of the output video. This resets the timestamps of the output video to start from 00:00:00. Otherwise, the video would start from 00:00:03 and the first three seconds would be black screens.

```python
import ffmpeg

in_file = ffmpeg.input(
    "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
)
ffmpeg.output(
    in_file.trim(start=5, end=8).setpts(expr="PTS-STARTPTS"),
    in_file.atrim(start=5, end=8).asetpts(expr="PTS-STARTPTS"),
    filename="trim_with_complex_filter.mp4",
).overwrite_output().run()

```

https://github.com/livingbio/typed-ffmpeg/assets/432851/a1f07e29-5c70-40b4-ae0a-a8dd9993a66c
